opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,27 +12,163 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# type: ignore
+
 from sys import version_info
+from typing import Dict, Tuple, Union
+
+# This is a cache to avoid going through creating a dictionary for all entry
+# points for 3.7 every time the entry points function is called.
+result_entry_points_37 = {}
 
 # FIXME remove this when support for 3.7 is dropped.
 if version_info.minor == 7:
     # pylint: disable=import-error
-    from importlib_metadata import entry_points, version  # type: ignore
+    from importlib_metadata import EntryPoint
+    from importlib_metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+    from importlib_metadata import version
+
+    def entry_points(
+        group: str = None, name: str = None
+    ) -> Union[Tuple, Dict]:
+
+        if group is None and name is None:
+
+            if not result_entry_points_37:
+
+                for entry_point in importlib_metadata_entry_points():
+
+                    if entry_point.group not in result_entry_points_37.keys():
+                        result_entry_points_37[entry_point.group] = []
+
+                    (
```

Comment:
Nit: What are these braces for?

Reply:
Actually nothing, made a mistake while making that line shorter than 80 characters, removin them :+1: